client.go

```diff
@@ -224,29 +246,44 @@ func NewClient(cfg *Config) (Client, error) {
 		return nil, fmt.Errorf("Valid db names are: %s and %s", DBNB, DBSB)
 	}
 
-	imp := &ovndb{
+	ovndb := &ovndb{
 		cache:        make(map[string]map[string]libovsdb.Row),
 		signalCB:     cfg.SignalCB,
 		disconnectCB: cfg.DisconnectCB,
 		db:           db,
+		addr:         cfg.Addr,
+		tlsConfig:    cfg.TLSConfig,
+		reconn:       cfg.Reconnect,
 	}
 
-	c, err := libovsdb.Connect(cfg.Addr, cfg.TLSConfig)
+	err := connect(ovndb)
 	if err != nil {
 		return nil, err
 	}
-	imp.client = c
-
-	initial, err := imp.client.MonitorAll(db, "")
-	if err != nil {
-		return nil, err
-	}
-
-	imp.populateCache(*initial)
-	notifier := ovnNotifier{imp}
-	imp.client.Register(notifier)
-
-	return imp, nil
+	return ovndb, err
+}
+
+func (c *ovndb) reconnect() {
+	ticker := time.NewTicker(500 * time.Millisecond)
```

Comment:
@hzhou8 : Can we move this logic in libovsdb client https://github.com/eBay/libovsdb/blob/master/client.go#L59 with Reconnect as a new function like libovsdb.Reconnect ? The thing is we will end up re-writing this same reconnect client logic for libovsdb too as this problem is common for ovs/ovn dbs.

Reply:
I did it first in libovsdb, but then I realized that it is not the right place because the IDL cache is maintained in go-ovn only.